### PR TITLE
Show the Dangers of a Re-org

### PR DESCRIPTION
**Pull request by hayesgm**
This patch adds a test that currently shows the danger of our a re-org. We will use this as one of the bases to test that our new work on re-org resistance is effective in real-life scenarios.

### CODE REVIEW

**Review by waynenilsen — APPROVED** [review]
very nice, working well.

I wonder if we want to write these up so they're failing somehow today since this is ..... not expected behavior or behavior that we would like to correct/change? In the spirit of TDD something to the effect of "step 1 write a failing test". Feel free to leave it, either way we will need to refactor just seems odd to see ✅ on "reorg allows double spend" expectation lol.

File: integration/__tests__/reorg_scen.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+      // Now it's time for the re-org
+      await eth.restore(snapshotId);
+      await ashley.tokenTransfer(bert, 100, usdc);
+      await eth.mine(20);
+
+      // Uh-oh
+      expect(await ashley.tokenBalance(usdc)).toEqual(0);
+      expect(await ashley.chainBalance(usdc)).toEqual(100);
+      expect(await starport.tokenBalance(usdc)).toEqual(0);
```
Comment (jflatow) [inline integration/__tests__/reorg_scen.js:35]: 💪

File: integration/util/scenario/chain.js [hunk 2]
```diff
@@ -181,6 +181,12 @@ class Chain {
     });
   }
 
+  async tokenBalance(token, chainAccount) {
+    console.log(["gghy88", token.toChainAsset(), chainAccount]);
```
Comment (jflatow) [inline integration/util/scenario/chain.js]: ?

**Review by jflatow — APPROVED** [review]
Very impressive, one question

**Comment by jflatow** [comment]
> very nice, working well.
> 
> I wonder if we want to write these up so they're failing somehow today since this is ..... not expected behavior or behavior that we would like to correct/change? In the spirit of TDD something to the effect of "step 1 write a failing test". Feel free to leave it, either way we will need to refactor just seems odd to see ✅ on "reorg allows double spend" expectation lol.

No matter what this is going to be the case right, and this test should pass (after re-org, starport doesn't have assets and transfer succeeds)? I think the breaking test we would want is that after the re-org, the Gateway state reflects the new reality, rather than continuing to debit/credit the funds

File: integration/util/scenario/chain.js (hunk 2 above)
Comment (hayesgm) [reply integration/util/scenario/chain.js]: FWIW, there are so many logs in integration tests that I use random strings so I can easily "find" a message in the clutter 🤦

**Comment by hayesgm** [comment]
I'll mark these as `skip` right now. I'd rather them show the current behavior because it's clearly wrong. If we just checked that they don't double-account, then it might have been the case that it was a set-up bug, etc, if they were green.

**Comment by hayesgm** [comment]
Also, depending on how we "clawback" the actual expectations aren't clearly defined yet.

**Pull request merged**